In [1]:
pip install opencv-python

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.1.1 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
import cv2

# Reading in an image

### Getting image shape

In [3]:
image = cv2.imread('Broccoli.jpg')
print(image.shape)

(419, 575, 3)


Where 419 is the height, 575 is the width and 3 is the number of channels.

### Viewing an image

In [4]:
cv2.imshow("Broccoli", image)
cv2.waitKey(0)

-1

Using cv2.waitKey(0) stops the dialog from closing before we get to see the image. So we can see the image in a dialog with the title we put in the first argument.

# Video Reading and showing

In [2]:
import cv2

video = cv2.VideoCapture('Comedy.mp4')

while True:
    ret, frame = video.read()

    if not ret:
        break   # End of video

    cv2.imshow('frame', frame)

    # Press q to quit
    if cv2.waitKey(25) & 0xFF == ord('q'):
        break

video.release()
cv2.destroyAllWindows()

In [4]:
import cv2

cap = cv2.VideoCapture(1)   # 0 = default webcam

while True:
    ret, frame = cap.read()

    if not ret:
        break

    cv2.imshow("Webcam", frame)

    # Press q to quit
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()

# Changing the color space

In [5]:
image = cv2.imread('Broccoli.jpg')
gray_image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
cv2.imshow("Broccoli", gray_image)
cv2.waitKey(0)

-1

# Converting to HSV

A color space to note is HSV. We can create comparisions between image with it using histograms. Lets convert our image to this

In [8]:
import cv2

image = cv2.imread('Broccoli.jpg')
gray_image = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
cv2.imshow("Broccoli", gray_image)
cv2.waitKey(0)

-1

# Resizing images

Lets see how we can resize image. To do this we need to run the cv2.resize() function.



In [10]:
import cv2

image = cv2.imread('Broccoli.jpg')
small_image = cv2.resize(image, (300, 300)) # resize image
cv2.imshow("Broccoli", small_image)
cv2.waitKey(0)

-1

# Saving images

We can save an image by calling the cv2.imwrite() function. It takes the name of the image and the object to save. Lets see the code below

In [11]:
import cv2

image = cv2.imread('Broccoli.jpg')
small_image = cv2.resize(image, (300, 300))
cv2.imwrite("Broccoli_Small.png", small_image)

True

In [18]:
image02 = cv2.imread('Bird01.jpg')
cv2.imshow("Bird01", image02)
cv2.waitKey(0)

-1

# Basic Blur Example

In [19]:
import cv2

img = cv2.imread("Bird01.jpg")

blur = cv2.blur(img, (15,15))

cv2.imshow("Original", img)
cv2.imshow("Blur Image", blur)

cv2.waitKey(0)
cv2.destroyAllWindows()

# Better Blur (Gaussian Blur)

In [21]:
import cv2

img = cv2.imread("Bird01.jpg")

blur = cv2.GaussianBlur(img, (15,15), 0)

cv2.imshow("Blurred", blur)

cv2.waitKey(0)
cv2.destroyAllWindows()

In [22]:
import cv2

img = cv2.imread("Bird01.jpg")

blur = cv2.GaussianBlur(img, (31,31), 0)

cv2.imshow("Blurred", blur)

cv2.waitKey(0)
cv2.destroyAllWindows()